Forwarded: not-needed
Author: David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2019-12-31
Description: [PATCH] Recalculate panel box visibility on applet change
 This is a Debian only patch since it is primarily to support a Debian
 third-party applet budgie-appmenu-applet

---
 src/panel/panel.vala | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

--- a/src/panel/panel.vala
+++ b/src/panel/panel.vala
@@ -136,6 +136,17 @@
 			return false;
 		}
 
+		private void panel_box_visibility() {
+			Gtk.Box[] boxarray = { start_box, center_box, end_box };
+			foreach (var box in boxarray) {
+				bool visible = true;
+				if (box.get_children().length() == 0) {
+					visible = false;
+				}
+				box.set_visible(visible);
+			}
+		}
+
 		/**
 		* Force update the geometry
 		*/
@@ -366,6 +377,7 @@
 			/* bit of a no-op. */
 			update_sizes();
 			load_applets();
+			this.applets_changed.connect(this.panel_box_visibility);
 		}
 
 		void do_size_allocate() {
@@ -732,6 +744,7 @@
 
 			set_applets();
 			budge_em_left(alignment, position);
+			panel_box_visibility();
 		}
 
 		void add_applet(Budgie.AppletInfo? info) {
@@ -781,6 +794,7 @@
 			ulong id = info.notify.connect(applet_updated);
 			info.set_data("notify_id", id);
 			this.applet_added(info);
+			panel_box_visibility();
 
 			if (this.is_fully_loaded) {
 				return;
@@ -1534,6 +1548,8 @@
 		private bool show_panel() {
 			show_panel_id = 0;
 
+			panel_box_visibility();
+
 			if (!this.allow_animation) {
 				return false;
 			}
